source/common/http/http2/codec_impl.h

```diff
@@ -334,7 +346,11 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
   StreamImpl* getStream(int32_t stream_id);
   int saveHeader(const nghttp2_frame* frame, HeaderString&& name, HeaderString&& value);
   void sendPendingFrames();
-  void sendSettings(const Http2Settings& http2_settings, bool disable_push);
+  void sendSettings(const envoy::config::core::v3::Http2ProtocolOptions& http2_options,
+                    bool disable_push);
+  // Callback triggered when the peer's SETTINGS frame is received.
+  // NOTE: This is only used for tests.
+  virtual void onSettings(const nghttp2_settings&) {}
```

Comment:
For now maybe call this `onSettingsForTest`?

Reply:
Done.